netkan/netkan/webhooks/github_mirror.py

```diff
@@ -0,0 +1,315 @@
+        "CC-BY-ND-2.0"      : 'https://creativecommons.org/licenses/by-nd/2.0/',
+        "CC-BY-ND-2.5"      : 'https://creativecommons.org/licenses/by-nd/2.5/',
+        "CC-BY-ND-3.0"      : 'https://creativecommons.org/licenses/by-nd/3.0/',
+        "CC-BY-ND-4.0"      : 'https://creativecommons.org/licenses/by-nd/4.0/',
+        "CC0"               : 'https://creativecommons.org/publicdomain/zero/1.0/',
+        "CDDL"              : 'https://opensource.org/licenses/CDDL-1.0',
+        "CPL"               : 'https://opensource.org/licenses/cpl1.0.php',
+        "EFL-1.0"           : 'https://opensource.org/licenses/ver1_eiffel',
+        "EFL-2.0"           : 'https://opensource.org/licenses/EFL-2.0',
+        "Expat"             : 'https://opensource.org/licenses/MIT',
+        "MIT"               : 'https://opensource.org/licenses/MIT',
+        "GPL-1.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-1.0.en.html',
+        "GPL-2.0"           : 'http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html',
+        "GPL-3.0"           : 'http://www.gnu.org/licenses/gpl-3.0.en.html',
+        "LGPL-2.0"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.0.html',
+        "LGPL-2.1"          : 'https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html',
+        "LGPL-3.0"          : 'http://www.gnu.org/licenses/lgpl-3.0.en.html',
+        "GFDL-1.1"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.1.en.html',
+        "GFDL-1.2"          : 'http://www.gnu.org/licenses/old-licenses/fdl-1.2.html',
+        "GFDL-1.3"          : 'http://www.gnu.org/licenses/fdl-1.3.en.html',
+        "LPPL-1.0"          : 'https://latex-project.org/lppl/lppl-1-0.html',
+        "LPPL-1.1"          : 'https://latex-project.org/lppl/lppl-1-1.html',
+        "LPPL-1.2"          : 'https://latex-project.org/lppl/lppl-1-2.html',
+        "LPPL-1.3c"         : 'https://latex-project.org/lppl/lppl-1-3c.html',
+        "MPL-1.1"           : 'https://www.mozilla.org/en-US/MPL/1.1/',
+        "Perl"              : 'http://dev.perl.org/licenses/',
+        "Python-2.0"        : 'https://www.python.org/download/releases/2.0/license/',
+        "QPL-1.0"           : 'https://opensource.org/licenses/QPL-1.0',
+        "W3C"               : 'https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document',
+        "Zlib"              : 'http://www.zlib.net/zlib_license.html',
+        "Zope"              : 'http://old.zope.org/Resources/License.1',
+        "Unlicense"         : 'https://unlicense.org/UNLICENSE',
+    }
+
+    def __init__(self, filename):
+        self.filename = Path(filename)
+        self.contents = self.filename.read_text()
+        self._raw     = json.loads(self.contents)
+
+    def can_mirror(self):
+        return (
+            self._raw.get('kind', 'package') == 'package'
+            and self._raw['download_content_type'] in Ckan.mime_to_extension
+            and self.redistributable()
+        )
```

Comment:
True/False returns like this are difficult to parse and I don't know that they save anything compute wise. I think an if/else block that falls to return False or True (pending what's easier and makes sense) is clearer.

Also you can refer to class properties with `self`, ie `self.mime_to_extension`.

Reply:
Hmm, I have to admit that I like this style. It has simpler control flow (one statement with short-circuiting), and it's self-documenting, since the conditions that are needed are listed one after the next, without extra `return` statements or multiple `if` conditions. And if a new condition comes up it's easy to add it to the list. It gives some of the benefits of functional programming, in that it avoiding side effects and mutable variables in favor of a declarative approach.

Reply:
I think being a Python project we should probably lean towards common patterns used by the language. Though I also don't work with functional languages that much and we don't have a style guide.

Reply:
I'm pretty sure `return` statements and the `and` operator are common patterns used by Python. :wink: 

I understand that you feel protective towards this code since it's your baby, but I can only reassure you that other contributors also want only the best for it, and sometimes something that you're not familiar with can turn out to be a good thing. Can we table this till all the code is actually working?

Reply:
Haha, that is true. That particular pattern is a touch foreign to me, but that's life and learning :slightly_smiling_face:, Long as we've got some test cases for it, we can always refactor if it becomes a maintenance burden. Though referring to class constants via the class name within the class, that I have a far stronger opinion about :wink: 

Absolutely, there are much bigger architectural decisions that are far more important. This was definitely a nit pick that got carried away. It's not that I'm super protective of the code per se, but I'm generally thinking about future me having to debug a production problem. Which from experience is never after an 8 hour sleep and just the right amount of coffee :smiley: 

As always, appreciate your effort and opinions!